js/ui/codingManager.js

```diff
@@ -473,9 +473,19 @@ const CodingManager = new Lang.Class({
     },
 
     _overviewHiding: function(overview, session) {
-        session.buttonApp.show();
-        if (session.buttonBuilder)
-            session.buttonBuilder.show();
+        let focusedWindow = global.display.get_focus_window();
+        if (!focusedWindow)
+            return;
+
+        let builderWindow = null;
+        if (session.actorBuilder)
+            builderWindow = session.actorBuilder.meta_window;
+
+        if (builderWindow && builderWindow === focusedWindow)
```

Comment:
Can just compare against `session.actorBuilder.window` since `focusedWindow` is guaranteed to be non-null